Util/channel/src/main/java/io/deephaven/util/channel/SeekableChannelsProvider.java

```diff
@@ -18,31 +18,80 @@
 public interface SeekableChannelsProvider extends SafeCloseable {
 
     /**
-     * Take the file source path or URI and convert it to a URI object.
+     * Take the file source path or URI string and convert it to a URI object.
      *
      * @param source The file source path or URI
+     * @param isDirectory Whether the source is a directory
      * @return The URI object
      */
-    static URI convertToURI(final String source) {
+    static URI convertToURI(final String source, final boolean isDirectory) {
+        if (source.isEmpty()) {
+            throw new IllegalArgumentException("Cannot convert empty source to URI");
+        }
         final URI uri;
         try {
             uri = new URI(source);
         } catch (final URISyntaxException e) {
             // If the URI is invalid, assume it's a file path
-            return new File(source).toURI();
+            return convertFileToURI(new File(source), isDirectory);
         }
         if (uri.getScheme() == null) {
-            // Need to convert to a "file" URI
-            return new File(source).toURI();
+            // Convert to a "file" URI
+            return convertFileToURI(new File(source), isDirectory);
         }
         return uri;
     }
 
+    /**
+     * Takes a file and convert it to a URI object. We should use this method instead of {@link File#toURI()} because
+     * {@link File#toURI()} internally calls {@link File#isDirectory()}, which can lead to disk access.
```

Comment:
```suggestion
     * Takes a file and convert it to a URI object. This method is preferred instead of {@link File#toURI()} because
     * {@link File#toURI()} internally calls {@link File#isDirectory()}, which typically invokes the {@code stat} system call, resulting in filesystem metadata access.
```